iphone/cli/commands/_build.js

```diff
@@ -3403,18 +3410,16 @@ iOSBuilder.prototype.writeEntitlementsPlist = function writeEntitlementsPlist()
 	// if we have a provisioning profile, make sure some entitlement settings are correct set
 	if (pp) {
 		// attempt to customize it by reading provisioning profile
+		if (!plist.hasOwnProperty('application-identifier')) {
+			plist['application-identifier'] = pp.appPrefix + '.' + this.tiapp.id;
+		}
+		plist['aps-environment'] = this.target === 'dist-appstore' || this.target === 'dist-adhoc' ? 'production' : 'development';
```

Comment:
OK, so we could do this: 

```javascript
if (pp.apsEnvironment) {
    plist['aps-environment'] = this.target === 'dist-appstore' || this.target === 'dist-adhoc' ? 'production' : 'development';
}
```

If they use APS, then we set it, but we set it to values based on the target, not the value in the provisioning profile.